### PR TITLE
ceph-*-build: do not build new branches on trusty

### PR DESCRIPTION
**Pull request by tchaikov**
by "new branches", i mean any branche whose name is not started with
"jewel" or "luminous".

see http://groovy-lang.org/operators.html, and
https://issues.jenkins-ci.org/browse/JENKINS-25465

Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
@alfredodeza if the `${GIT_BRANCH, fullName=false}` trick does not work, i will need to add a shell script to export a new env variable and use that one instead.

File: ceph-build/config/definitions/ceph-build.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@
       - github:
           url: https://github.com/ceph/ceph
     execution-strategy:
-       combination-filter: DIST==AVAILABLE_DIST && ARCH==AVAILABLE_ARCH && (ARCH=="x86_64" || (ARCH == "arm64" && (DIST == "xenial" || DIST == "centos7")))
+       combination-filter: DIST == AVAILABLE_DIST && ARCH == AVAILABLE_ARCH && (ARCH=="x86_64" || (ARCH == "arm64" && ["xenial", "centos7"].contains(DIST))) && (${GIT_BRANCH, fullName=false} ==~ /(jewel|luminous).*/ || (! ${GIT_BRANCH, fullName=false} ==~ /(jewel|luminous).*/ && DIST != "trusty")))
```
Comment (alfredodeza) [inline ceph-build/config/definitions/ceph-build.yml]: have you tested any of this? this looks super scary

Comment (tchaikov) [reply ceph-build/config/definitions/ceph-build.yml]: @alfredodeza i tested using my own jenkins instance. but i don't have the necessary privilege for testing my change on jenkins.ceph.com , so i haven't test this PR using it yet. i mailed you my API token in hope that i will be able to test it.

File: ceph-dev-new-trigger/config/definitions/ceph-dev-new-trigger.yml [hunk 2]
```diff
@@ -28,20 +28,46 @@
...
-        - project: 'ceph-dev-new'
-          predefined-parameters: |
-            BRANCH=${GIT_BRANCH}
-            FORCE=True
-        - project: 'ceph-dev-new'
-          predefined-parameters: |
-            BRANCH=${GIT_BRANCH}
-            FORCE=True
-            FLAVOR=notcmalloc
+      - conditional-step:
```
Comment (tchaikov) [inline ceph-dev-new-trigger/config/definitions/ceph-dev-new-trigger.yml]: @alfredodeza i switched to "Conditional BuildStep Plugin". seems `combination-filter` does not allow me to access env variables. hopefully this is less scary, and works. but i am not sure if this plugin is installed in our jenkins. could you help me verify it?

Comment (alfredodeza) [reply ceph-dev-new-trigger/config/definitions/ceph-dev-new-trigger.yml]: Ok so this will need that plugin installed. /cc @djgalloway

Comment (djgalloway) [reply ceph-dev-new-trigger/config/definitions/ceph-dev-new-trigger.yml]: @alfredodeza It's already installed.  Green light from me as far as that goes.

**Comment by tchaikov** [comment]
@alfredodeza tested by updating jenkins.ceph.com with the new trigger jobs. 

- see https://jenkins.ceph.com/job/ceph-dev/14152/parameters/ and https://jenkins.ceph.com/job/ceph-dev/14153/parameters/ . the "DISTROS" parameter of "ceph-dev" jobs triggered by [mimic-dev1](https://jenkins.ceph.com/job/ceph-dev-trigger/7442/) push is now all "xenial centos7".
- see https://jenkins.ceph.com/job/ceph-dev/14150/parameters/ and https://jenkins.ceph.com/job/ceph-dev/14151/parameters/ . the "DISTROS" parameter of "ceph-dev" jobs triggered by [wip-19257-luminous](https://jenkins.ceph.com/job/ceph-dev-trigger/7442/) pushed to ceph/ceph-ci is now all "xenial centos7" is still "xenial trusty centos7"

File: ceph-dev-new-trigger/config/definitions/ceph-dev-new-trigger.yml [hunk 3]
```diff
@@ -28,20 +28,47 @@
...
-      - trigger-builds:
-        - project: 'ceph-dev-new'
-          predefined-parameters: |
-            BRANCH=${GIT_BRANCH}
-            FORCE=True
-        - project: 'ceph-dev-new'
-          predefined-parameters: |
-            BRANCH=${GIT_BRANCH}
-            FORCE=True
-            FLAVOR=notcmalloc
```
Comment (alfredodeza) [inline ceph-dev-new-trigger/config/definitions/ceph-dev-new-trigger.yml]: I know this is already merged, but it seems that `FLAVOR=notcmalloc` was not kept with the addition. This means identical builds

Comment (tchaikov) [reply ceph-dev-new-trigger/config/definitions/ceph-dev-new-trigger.yml]: @alfredodeza fixed at #938.

**Pull request merged**